lib/std/http/status.zig

```diff
@@ -0,0 +1,203 @@
+                .already_reported => "Already Reported",
+                .im_used => "IM Used",
+
+                .multiple_choice => "Multiple Choice",
+                .moved_permanently => "Moved Permanently",
+                .found => "Found",
+                .see_other => "See Other",
+                .not_modified => "Not Modified",
+                .use_proxy => "Use Proxy",
+                .temporary_redirect => "Temporary Redirect",
+                .permanent_redirect => "Permanent Redirect",
+
+                .bad_request => "Bad Request",
+                .unauthorized => "Unauthorized",
+                .payment_required => "Payment Required",
+                .forbidden => "Forbidden",
+                .not_found => "Not Found",
+                .method_not_allowed => "Method Not Allowed",
+                .not_acceptable => "Not Acceptable",
+                .proxy_auth_required => "Proxy Authentication Required",
+                .request_timeout => "Request Timeout",
+                .conflict => "Conflict",
+                .gone => "Gone",
+                .length_required => "Length Required",
+                .precondition_failed => "Precondition Failed",
+                .payload_too_large => "Payload Too Large",
+                .uri_too_long => "URI Too Long",
+                .unsupported_media_type => "Unsupported Media Type",
+                .range_not_satisfiable => "Range Not Satisfiable",
+                .expectation_failed => "Expectation Failed",
+                .teapot => "I'm a teapot",
+                .misdirected_request => "Misdirected Request",
+                .unprocessable_entity => "Unprocessable Entity",
+                .locked => "Locked",
+                .failed_dependency => "Failed Dependency",
+                .too_early => "Too Early",
+                .upgrade_required => "Upgrade Required",
+                .precondition_required => "Precondition Required",
+                .too_many_requests => "Too Many Requests",
+                .header_fields_too_large => "Request Header Fields Too Large",
+                .unavailable_for_legal_reasons => "Unavailable For Legal Reasons",
+
+                .internal_server_error => "Internal Server Error",
+                .not_implemented => "Not Implemented",
+                .bad_gateway => "Bad Gateway",
+                .service_unavailable => "Service Unavailable",
+                .gateway_timeout => "Gateway Timeout",
+                .http_version_not_supported => "HTTP Version Not Supported",
+                .variant_also_negotiates => "Variant Also Negotiates",
+                .insufficient_storage => "Insufficient Storage",
+                .loop_detected => "Loop Detected",
+                .not_extended => "Not Extended",
+                .network_authentication_required => "Network Authentication Required",
+            };
+        }
+    };
+
+    pub const Unknown = struct {
+        code: u16,
+        phrase: []const u8,
```

Comment:
Note that phrases are deprecated in HTTP. They do not exist since http2; so this would at least need to be optional.

Reply:
why not always leave it as empty string then in that mode? because its not optional in 1.x

Reply:
Because an empty string is distinct from a missing string

Reply:
I'd recommend you stop the reason phrase entirely. See https://github.com/httpwg/http2-spec/issues/202#issuecomment-362946819

Or at least I don't think it belongs in the zig standard library.

Reply:
it's needed when constructing and parsing a response, which is going to be important when we add a server or client implementation down the line

Reply:
unless we don't want to support being able to parse a reason when using non-standard codes, which is fine, but a call we have to make

Reply:
> unless we don't want to support being able to parse a reason when using non-standard codes, which is fine, but a call we have to make

IMO the low level http1-specific method would do so; but not the higher level layers that normal users would use.

I also advise e.g. exposing method and status as "just another header".

Reply:
okay